app/Http/Controllers/Apis/Protected/Summit/OAuth2SummitEventsApiController.php

```diff
@@ -1549,4 +1548,108 @@ public function upgradeEvent($summit_id, $event_id, $type_id)
             );
         });
     }
+
+    /**
+     * @param $summit_id
+     * @return \Illuminate\Http\JsonResponse|mixed
+     */
+    public function getOverflowStreaming($summit_id)
+    {
+        return $this->processRequest(function () use ($summit_id) {
+            $query_string_key = config("overflow.query_string_key", "k");
+            if (!Request::has($query_string_key)) return $this->error404();
+
+            $summit = SummitFinderStrategyFactory::build($this->repository, $this->resource_server_context)->find($summit_id);
+            if (is_null($summit))
+                return $this->error404();
+
+            $overflow_stream_key = Request::get($query_string_key);
+
+            $event = $this->event_repository->getByOverflowStreamKey($overflow_stream_key);
+
+            if (is_null($event))
+                return $this->error404();
+
+            return $this->ok(SerializerRegistry::getInstance()
+                ->getSerializer($event, IPresentationSerializerTypes::OverflowStream)
+                ->serialize
+                (
+                    SerializerUtils::getExpand(),
+                    SerializerUtils::getFields(),
+                    SerializerUtils::getRelations(),
+                )
+            );
+        });
+    }
+
+    /**
+     * @param $summit_id
+     * @param $event_id
+     * @return mixed
+     */
+    public function updateOverflowInfo($summit_id, $event_id) {
+         return $this->processRequest(function() use($summit_id, $event_id){
+
+            Log::debug(sprintf("OAuth2SummitEventsApiController::updateOverflowInfo summit id %s event id %s", $summit_id, $event_id));
+
+            if (!Request::isJson()) return $this->error400();
+            $data = Request::json();
+
+            $summit = SummitFinderStrategyFactory::build($this->repository, $this->resource_server_context)->find($summit_id);
+            if (is_null($summit))
+                return $this->error404();
+
+            $event = $summit->getEvent($event_id);
+            if (is_null($event))
+                return $this->error404();
+
+            $payload = $data->all();
+
+            $validation = Validator::make($payload, [
```

Comment:
move this to
app/Http/Controllers/Apis/Protected/Summit/Factories/SummitEventValidationRulesFactory.php
create a new static method to validate overflow info
and remove overflow fields from the regular methds

Reply:
done